src/sentry/reprocessing2.py

```diff
@@ -275,24 +292,97 @@ def delete_old_primary_hash(event):
     If the primary_hash is not part of the PK/sortkey, or if the primary_hash
     did not change, nothing needs to be done as ClickHouse's table merge will
     merge the two rows together.
+
+    Like `buffered_handle_remaining_events` this is a quick and dirty way to
+    batch event IDs so requests to tombstone rows are not being individually
+    sent over to Snuba.
+
+    This also includes the same constraints for optimal performance as
+    `buffered_handle_remaining_events` in that events being fed to this
+    should have datetimes as close to each other as possible. The assumption
+    is that since this goes through the same pipeline as
+    `buffered_handle_remaining_events`, this requirement continues to be
+    fulfilled.
```

Comment:
this assumption could be completely wrong. tombstoning happens significantly later on in the reprocessing pipeline compared to `handle_remaining_events`. reprocessing hands off a lot of its work to asynchronous celery tasks, which means there's no guarantee that the datetimes of the events are even clustered together nearly as well as they were in `handle_remaining_...`. i don't know if there's any way for us to continue to maintain this prerequisite for tombstoning though.

Reply:
This comment is a good point and I would add it to the code comment. I think we'll just have to see.